File: Wox.UI.Electron/src/enums/WoxQueryTypeEnum.ts
import { BaseEnum } from "./base/BaseEnum.ts"

export type WoxQueryType = string

export class WoxQueryTypeEnum extends BaseEnum {
  static readonly WoxQueryTypeInput = WoxQueryTypeEnum.define("input", "input")
  static readonly WoxQueryTypeSelection = WoxQueryTypeEnum.define("selection", "selection")
}

File: Wox.UI.Flutter/wox/lib/enums/wox_query_type_enum.dart
typedef WoxQueryType = String;

enum WoxQueryTypeEnum {
  WOX_QUERY_TYPE_INPUT("input", "input"),
  WOX_QUERY_TYPE_SELECTION("selection", "selection");

  final String code;
  final String value;

  const WoxQueryTypeEnum(this.code, this.value);

  static String getValue(String code) => WoxQueryTypeEnum.values.firstWhere((activity) => activity.code == code).value;
}
